src/main/java/com/cst438/controller/AssignmentController.java

```diff
@@ -5,114 +5,275 @@
 import com.cst438.dto.AssignmentStudentDTO;
 import com.cst438.dto.GradeDTO;
 import org.springframework.http.HttpStatus;
+import org.springframework.security.access.prepost.PreAuthorize;
 import org.springframework.web.bind.annotation.*;
 import org.springframework.beans.factory.annotation.Autowired;
-import org.springframework.web.bind.annotation.CrossOrigin;
-import org.springframework.web.bind.annotation.PostMapping;
-import org.springframework.web.bind.annotation.RequestBody;
-import org.springframework.web.bind.annotation.RestController;
 import org.springframework.web.server.ResponseStatusException;
 
-
+import java.security.Principal;
+import java.text.ParseException;
+import java.text.SimpleDateFormat;
 import java.util.ArrayList;
+import java.util.Date;
 import java.util.List;
 
+
+
 @RestController
 @CrossOrigin(origins = "http://localhost:3000")
 public class AssignmentController {
 
+    @Autowired
+    private AssignmentRepository assignmentRepository;
+    @Autowired
+    private SectionRepository sectionRepository;
+    @Autowired
+    private GradeRepository gradeRepository;
+
+    @Autowired
+    private  UserRepository userRepository;
+
+    @Autowired
+    private EnrollmentRepository enrollmentRepository;
+
 
     // instructor lists assignments for a section.  Assignments ordered by due date.
     // logged in user must be the instructor for the section
+    // TEST URL http://localhost:8080/sections/8/assignments
+    @PreAuthorize("hasAuthority('SCOPE_ROLE_INSTRUCTOR')")
     @GetMapping("/sections/{secNo}/assignments")
-    public List<AssignmentDTO> getAssignments(
-            @PathVariable("secNo") int secNo) {
+    public List<AssignmentDTO> getAssignments(@PathVariable("secNo") int secNo, Principal principal)  {
+        String instructorEmail = principal.getName();
+        Section section = sectionRepository.findById(secNo).orElse(null);
+        if (section == null) {
+            throw new ResponseStatusException(HttpStatus.NO_CONTENT, "section not found. secNo: "+secNo);
+        } else if (section.getInstructorEmail() == null || !section.getInstructorEmail().equals(instructorEmail)) {
+            throw new ResponseStatusException(HttpStatus.UNAUTHORIZED, "unauthorized");
+        }
+
+        List<Assignment> assignments = assignmentRepository.findBySectionNoOrderByDueDate(secNo);
+        //we are not checking for empty/ null assignments
```

Comment:
I added this comment after doing a code review for another team who did the empty/ null check.  Do you guys think we should add a check? If ia check doesn't seem necessary, I can just delete the comment.

Reply:
Yeah I am not sure. It seems to return an empty list when a section has no assignments.